File: frontend/lib/features/admin/data/models/content_upload_request.dart
/// Request body of the `data` part on `POST/PUT /api/admin/shows|audios`
/// (`ContentUploadRequest` on the backend).
class ContentUploadRequest {
  const ContentUploadRequest({
    required this.title,
    this.description,
    this.categoryId,
    this.speaker,
    this.isPremium = false,
  });

  final String title;
  final String? description;
  final String? categoryId;
  final String? speaker;
  final bool isPremium;

  Map<String, dynamic> toJson() => {
        'title': title,
        'description': description,
        'categoryId': categoryId,
        'speaker': speaker,
        'isPremium': isPremium,
      };
}

File: audio/src/main/java/com/nostalgiaana/audio/content/dto/ContentUploadRequest.java
package com.nostalgiaana.audio.content.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

import java.util.UUID;

@Data
public class ContentUploadRequest {

    @NotBlank(message = "Title is required")
    private String title;

    private String description;

    private UUID categoryId;

    private String speaker;

    // Lombok generates isPremium()/setIsPremium() for a primitive boolean field
    // already named "is...", which Jackson resolves to two different property
    // names ("premium" vs "isPremium") and silently drops incoming values.
    // Pinning the JSON key here sidesteps that mismatch.
    @JsonProperty("isPremium")
    private boolean isPremium = false;
}
